### PR TITLE
Bump go 1.8

### PR DESCRIPTION
**Pull request by emilevauge**
This PR bumps go version to 1.8 and removes dependency to https://github.com/mailgun/manners using [native graceful shutdown](https://beta.golang.org/doc/go1.8#http_shutdown) instead.

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -490,21 +490,22 @@ func (server *Server) createTLSConfig(entryPointName string, tlsOption *TLS, rou
 	return config, nil
 }
 
-func (server *Server) startServer(srv *manners.GracefulServer, globalConfiguration GlobalConfiguration) {
+func (server *Server) startServer(srv *http.Server, globalConfiguration GlobalConfiguration) {
 	log.Infof("Starting server on %s", srv.Addr)
 	if srv.TLSConfig != nil {
-		if err := srv.ListenAndServeTLSWithConfig(srv.TLSConfig); err != nil {
+		err := srv.ListenAndServeTLS("", "")
```
Comment (timoreimann) [inline server.go]: You could declare the `err` variable at the top, assign it in each TLS/no-TLS case, and check for `nil` once only:

```go
var err error
if srv.TLSConfig != nil {
  err = srv.ListenAndServeTLS("", "")
} else {
  err = srv.ListenAndServe()
}

if err != nil {
  log.Fatal("Error creating server: ", err)
}
```

Saves us one nested level of `if`ing.

File: server.go [hunk 2]
```diff
@@ -114,14 +113,15 @@ func (server *Server) Wait() {
...
+	defer log.Info("Server stopped")
 	for serverEntryPointName, serverEntryPoint := range server.serverEntryPoints {
 		ctx, cancel := context.WithTimeout(context.Background(), time.Duration(server.globalConfiguration.GraceTimeOut)*time.Second)
-		go func() {
-			log.Debugf("Waiting %d seconds before killing connections on entrypoint %s...", 30, serverEntryPointName)
-			serverEntryPoint.httpServer.BlockingClose()
-			cancel()
-		}()
-		<-ctx.Done()
+		log.Debugf("Waiting %d seconds before killing connections on entrypoint %s...", server.globalConfiguration.GraceTimeOut, serverEntryPointName)
```
Comment (timoreimann) [inline server.go]: What's the reason we don't wrap this in a go routine anymore?

IIUC, we are now shutting down the server entry points sequentially whereas we used to do so concurrently.

Comment (emilevauge) [inline server.go]: In fact, it was not working in the old version because of `<-ctx.Done()`.
I used a `WaitGroup` instead.

File: server.go (hunk 1 above)
Comment (emilevauge) [inline server.go]: Yep, done.

**Comment by timoreimann** [comment]
Updated and green!

We need one more LGTM. :-)

**Pull request merged**